### PR TITLE
fixes #8489 - set KeepAlive On in foreman vhost

### PR DESCRIPTION
**Pull request by mmoll**
As it seems to be still a longer time to puppetlabs-apache 2.0, handle this in the vhost.
@lzap for your testing pleasure... :snail:

### CODE REVIEW

**Comment by lzap** [comment]
LGTM but I don't have enough confidence to merge Puppet stuff. Also, I think we should parametrize MaxKeepAliveRequests and KeepAliveTimeout options.

**Comment by mmoll** [comment]
@lzap like that?
tests are still missing...

**Comment by lzap** [comment]
That looks good, you can also parametrize keep-alive itself.

**Comment by mmoll** [comment]
@lzap done :gift:
Now also with a test.

File: manifests/init.pp [hunk 1]
```diff
@@ -246,6 +257,9 @@
   $server_ssl_certs_dir      = $::foreman::params::server_ssl_certs_dir,
   $server_ssl_key            = $::foreman::params::server_ssl_key,
   $server_ssl_crl            = $::foreman::params::server_ssl_crl,
+  $keepalive                 = $::foreman::params::keepalive,
```
Comment (ekohl) [inline manifests/init.pp]: Call validate_*() on these parameters?

**Comment by ekohl** [comment]
:+1: when the validation functions are used.

File: manifests/init.pp (hunk 1 above)
Comment (mmoll) [inline manifests/init.pp]: I do in manifests/config/passenger.pp

Comment (ekohl) [inline manifests/init.pp]: Touche

File: manifests/init.pp [hunk 2]
```diff
@@ -132,6 +132,17 @@
...
 # $server_ssl_crl::             Defines the Apache mod_ssl SSLCARevocationFile setting in Foreman vhost conf file.
 #
+# $keepalive::                  Enable KeepAlive setting of Apache?
+#                               type:boolean
+#
+# $maxkeepaliverequests::       MaxKeepAliveRequests setting of Apache
+#                               (Number of requests allowed on a persistent connection)
+#                               type:integer
+#
+# $keepalivetimeout::           KeepAliveTimeout setting of Apache
```
Comment (domcleal) [inline manifests/init.pp]: Should these parameters contain underscores separating the words?  Other parameters do, e.g. `passenger_start_timeout`

Comment (mmoll) [inline manifests/init.pp]: changed

**Comment by domcleal** [comment]
:+1: when tests pass.  I tested the earlier version (prior to the parameter rename) successfully.

**Comment by domcleal** [comment]
Merged, thanks @mmoll.

**Pull request merged**